patoolib/__init__.py

```diff
@@ -382,20 +395,22 @@ def list_formats ():
                       (command, util.strlist_with_or(handlers)))
 
 
-def check_program_compression(archive, command, program, compression):
+def check_program_compression(program, compression):
     """Check if a program supports the given compression."""
     program = os.path.basename(program)
     if compression:
-        # check if compression is supported
+        # check if compression is supported natively
         if not program_supports_compression(program, compression):
-            if command == 'create':
-                comp_command = command
-            else:
-                comp_command = 'extract'
-            comp_prog = find_archive_program(compression, comp_command)
-            if not comp_prog:
-                msg = "cannot %s archive `%s': compression `%s' not supported"
-                raise util.PatoolError(msg % (command, archive, compression))
+            # Check if compression is supported via an external program
+            # Note that we expect the program name to be identical to the
+            # compression type
+            if program in ('tar', 'star', 'bsdtar'):
+                comp_prog = util.find_program(compression)
+                if comp_prog:
+                    return True
+            return False
+
+    return True
```

Comment:
so if compression is None (or other False) it would now return True and no exception is raised in either case.
I am not familiar yet with code/logic enough yet on the use/assumptions but sounds a bit odd, but shouldn't it return False or None?  in either case -- docstring should be adjusted.

Also since it is a part of public API, this change could break someone's code which relied on catching the exception... 
 I guess ideally this function "signature" should remain the same but new code should go into a new function like  `is_compression_supported` which would be used within RFed `check_program_compression` and through the code where desired to be used instead of `check_program_compression`.

Reply:
I didn't think it was part of the public API at the time as it's not listed in `__all__`, but I could certainly rewrite it to keep the interface the same. 

Also, looking at it again, I see that in the new place I've used it on line 346, compression would never have a value of None, so the check should be removed. The idea would be that this returns True if the given compression type is supported by the program, and False otherwise.

Reply:
This should be fixed now